### DISCUSSION TITLE
Need help on below issue

### DISCUSSION THREAD

**Discussion by Nikkihere** [sassoftware/saspy#553] [General]
There are 2 things which i wanted to discuss

**_First_** 
As mentioned in the #550, I have somewhat similar logic, where I am reading the large file around 7GB in chunks and do some data manipulation and eventually save each chunk as sas table but after processing around 7 iterations the system does not write further files and keeps on running but when interrupted i get a below error. 

**ValueError: invalid literal for int() with base 10: 'No SAS process attached. SAS process has terminated unexpectedly.**

Looking at error message it might be related too #207 

Guess this might be related to data so PFB details of data and log

The input sas file has around 1000 columns and 40 millions records where it has many date fields having default value as 9999-01-01(which may be out of date range for data frame)
and also has a lot of columns containing values that are blank or ".".

PFB details

SASsession Object
Access Method = IOM
SAS Config name = winlocal
SAS Config file = C:\Users\Nikki\Desktop\Scenario\venv\Lib\site-packages\saspy\sascfg_personal.py
WORK Path = 
SAS Version = 9.04.01M7P08062020
SASPy Version = 5.2.2
Teach me SAS = False
Batch = False
Results = Pandas
SAS Session Encoding = wlatin1
Python Encoding value = windows-1252
SAS process Pid value = 17589
a
when interrupted, i see the below last log
![IMG_20230729_200539](https://github.com/sassoftware/saspy/assets/140905421/3d2a7b01-dc6b-4f2c-a0e3-5e4dba7bd1b1)

**Also does using classpath in config might cause this issue? I read somewhere that classpath is not needed now and should be removed as it might cause issues**

**_Second_**
when I am submitting the below sas code from python, it does not create an output file and just keeps on running.

```
sascode = ''' 
 LIBNAME INLIB 'C:\Users\Nikki\Desktop\input';
 LIBNAME OUTLIB 'C:\Users\Nikki\Desktop\output';
 
 PROC CONTENTS data = INLIB.TEST1 out = OUTLIB.TESTOUT;
 RUN;
 '''

results = sas.submit(sas_code)
```

**Comment by tomweber-sas** [comment]
Hey, I can help with this. I think the first thing to do is based upon this question: 
`Also does using classpath in config might cause this issue? `

Yes, it certainly could cause problems. You shouldn't be coding your own classpath; it does say that in the instructions for configuring the IOM access method: https://sassoftware.github.io/saspy/configuration.html#attn-as-of-saspy-version-3-3-3-the-classpath-is-no-longer-required-in-your-configuration-file

Did you try removing your classpath and see if that made a difference? Can you show what your configuration is, so I could have a better idea? 

I don't know what code you're running for the data you mentioned. That proc contents ought to run and finish, so I don't know what's going on with that or how you're determining what's happening. I would need more info. But lets start with your configuration and get rid of the classpath you coded, and go from there.

Tom

**Comment by Nikkihere** [comment]
Thanks for the response.

**Regarding First issue:**
PFA my config file. I have now removed the class path but still face the same issue and error.
[sascfg_personal.txt](https://github.com/sassoftware/saspy/files/12231294/sascfg_personal.txt)

**Regarding Second issue:**
I somehow got what is causing this but not sure whether this is a issue or am I missing something but if I provide the path with folder name as numeric(like here 202112) then it does not create an output file and just keeps on running but when I change the folder name to say(Input or Temp) it works without any issue.

```
sascode = ''' 
 LIBNAME INLIB 'C:\Users\Nikki\Desktop\202112';    <---
 LIBNAME OUTLIB 'C:\Users\Nikki\Desktop\output';
 
 PROC CONTENTS data = INLIB.TEST1 out = OUTLIB.TESTOUT;
 RUN;
 '''

results = sas.submit(sas_code)
```

**Comment by tomweber-sas** [comment]
Hey, looking at the config file, it still has the classpath coded. What you're describing above about having a path with a number doesn't really make any sense. Would need to see the SAS Log to know what's going on. I can't reproduce anything like that.

I tried the code above, and it has syntax errors. Accounting for those, it runs as I would expect. But that begs the question of what you actually ran and what it did. Without seeing anything but source code snippets, there's nothing I can really tell. I need to see actual runs of the code and output and logs. 

Can you run this and show the whole run, including output? Maybe there will be something I can see

```
import saspy
sas = saspy.SASsession(cfgname='winlocal')
sas

sas.sascfg.SAScfg.winlocal

sas._io.sascfg.classpath

sas.saslib('inlib', path=r'c:\Users\Nikki\Desktop\202112')

sas.saslib('outlib', path=r'c:\Users\Nikki\Desktop\output')

sas.submitLOG('proc contents data=inlib.test1 out=outlib.testout;run;')

print(sas.saslog())
```
